smolyay/samples.py

```diff
@@ -0,0 +1,207 @@
+import abc
+import sys
+import numpy
+
+
+class UnidimensionalPointSet(abc.ABC):
+    """Set of unidimensional points
+
+    A set of unique unidimensional points to be used as the sampling points
+    in a tensor product grid or sparse grid. These unidimensional points are
+    associated with certain polynomial families, and the combination of
+    unique points and polynomial families are the basis for different types
+    of quadrature rules that are used in numerical integration and in
+    approximation.
+    The natural domain of the points specified by :param:``natural_domain`` is
+    the domain in which all the possible points in the set lies within.
+    ``number_points`` is the number of points that the UnidimensionalPointSet
+    is expected to store.
+    ``domain`` is the domain of the points generated by the point set.
+    """
+
+    def __init__(self):
+        self._points = []
```

Comment:
Default to not creating them
```suggestion
        self._points = None
```